packages/history/history.ts

```diff
@@ -0,0 +1,983 @@
+/**
+ * A URL path including the pathname, search string, and hash. No URL
+ * protocol or domain information should be part of a path.
+ */
+export type Path = string;
+
+/**
+ * The pieces of a URL path.
+ */
+export type PathPieces = {
+  /**
+   * The URL pathname, beginning with a /.
+   */
+  pathname?: string;
+
+  /**
+   * The URL search string, beginning with a ?.
+   */
+  search?: string;
+
+  /**
+   * The URL fragment identifier, beginning with a #.
+   */
+  hash?: string;
+};
+
+/**
+ * An object that is used to associate some arbitrary data with a
+ * location, but that does not appear in the URL path.
+ */
+export type State = object;
```

Comment:
Hey, not sure if you saw this, but I left a comment in the original [TS PR](https://github.com/ReactTraining/history/pull/774) about whether this type is correct: https://github.com/ReactTraining/history/pull/774#discussion_r376315739